### PR TITLE
Closes #71: Make JMapWrapper final class

### LINKED ISSUE #71: JMapWrapper should be serializable
Otherwise it won't work with Spark.

### PR DESCRIPTION
**Pull request by regadas**

### CODE REVIEW

File: core/src/main/scala/com/spotify/featran/FeatureBuilder.scala [hunk 1]
```diff
@@ -274,11 +274,14 @@ object FeatureBuilder {
 
 }
 
-private class JMapWrapper[K, V](val m: java.util.Map[K, V]) extends Map[K, V] {
+private final class JMapWrapper[K, V](m: java.util.Map[K, V]) extends Map[K, V] with Serializable {
```
Comment (nevillelyh) [inline core/src/main/scala/com/spotify/featran/FeatureBuilder.scala]: It's not a case class though, confusing commit message?

**Comment by nevillelyh** [comment]
Do we need a test for this? IIRC it affects `featran-spark` when we extract features as `Map[String, T]`.

File: core/src/main/scala/com/spotify/featran/FeatureBuilder.scala (hunk 1 above)
Comment (regadas) [reply core/src/main/scala/com/spotify/featran/FeatureBuilder.scala]: @nevillelyh ah yeah updated

**Comment by regadas** [comment]
@nevillelyh hm, yeah let's do that

**Comment by regadas** [comment]
@nevillelyh test added

File: core/src/test/scala/com/spotify/featran/Fixtures.scala [hunk 2]
```diff
@@ -43,6 +43,14 @@ object Fixtures {
                            Seq(0.0, 0.0, 0.0, 1.0, 0.0, 0.75),
                            Seq(0.0, 0.0, 0.0, 0.0, 1.0, 1.00))
 
+  val expectedMapValues = Seq(
+    Map("min_max" -> 0.0, "one_hot_a" -> 1.0),
+    Map("min_max" -> 0.25, "one_hot_b" -> 1.0),
+    Map("one_hot_c" -> 1.0, "min_max" -> 0.5),
```
Comment (nevillelyh) [inline core/src/test/scala/com/spotify/featran/Fixtures.scala]: purely aesthetic, sort the keys so it's more readable?

**Pull request merged**